worldguard-legacy/src/main/java/com/sk89q/worldguard/protection/flags/DefaultFlag.java

```diff
@@ -179,4 +189,36 @@ private DefaultFlag() {
 
         return null;
     }
+
+    /**
+     * Add a new custom flag to the flags list
+     *
+     * @param flag The flag to add
+     */
+    static void addFlag(Flag<?> flag) {
+        Flag<?> match = fuzzyMatchFlag(flag.getName());
+        if (match != null) {
+            throw new IllegalArgumentException("Duplicate flag");
+        }
+        Flag<?>[] newList = Arrays.copyOf(flagsList, flagsList.length + 1);
+        newList[flagsList.length] = flag;
+
+        // Force update the flagsList
+        // flagsList has to be public to allow other packages to access it, yet
+        // we have to be able to change it too :/
+        try {
+            java.lang.reflect.Field field = DefaultFlag.class.getField("flagsList");
+            if (!addFlagsUnlocked) {
+                addFlagsUnlocked = true;
+                java.lang.reflect.Field modifiersField = java.lang.reflect.Field.class.getDeclaredField("modifiers");
+                modifiersField.setAccessible(true);
+                modifiersField.setInt(field, field.getModifiers() & ~java.lang.reflect.Modifier.FINAL);
+            }
+            field.set(null, newList);
```

Comment:
An example of a plugin that uses this is here:
https://github.com/NerdNu/NerdFlags/blob/master/src/com/michaelelin/NerdFlags/NerdFlagsPlugin.java